onnxscript/function_libs/torch_aten/ops/core.py

```diff
@@ -1487,7 +1487,7 @@ def aten_cumprod_backward(
     raise NotImplementedError()
 
 @torch_op("aten::cumsum", trace_only=True)
-def aten_cumsum(self: TRealUnlessInt16OrInt8, dim: int, dtype: int = -1) -> TRealUnlessInt16OrInt8:
+def aten_cumsum(self: TRealUnlessInt16OrInt8, dim: Union[INT32, INT64], dtype: int = -1) -> TRealUnlessInt16OrInt8:
```

Comment:
Can dim be INT32 (supported by cumsum)? Do we need (a) an explicit cast to INT64, or (b) just assume it’s INt64 and let caller handle type conversion?

Reply:
Ahh, I get it now. Making a subfunction covering as much as content we could from the origin enables a relative compact function layer still, instead of tracing the whole aten function? Which means the curren solution for dtype if branch would be also like this?

Reply:
> Can dim be INT32 (supported by cumsum)? Do we need (a) an explicit cast to INT64, or (b) just assume it’s INt64 and let caller handle type conversion?

Get this from https://github.com/onnx/onnx/blob/main/docs/Changelog.md#CumSum-14

T2 accepts both INT32 and INT64 though

Reply:
> Ahh, I get it now. Making a subfunction covering as much as content we could from the origin enables a relative compact function layer still, instead of tracing the whole aten function? Which means the curren solution for dtype if branch would be also like this?

Precisely! What do you think?

Reply:
I think it's great for a temp solution, as at the end of the day, it's a onnx-script issue to support these pywriting anyway. cc @BowenBao This is the discussion about how to deal with unsupported control flow.